ci-operator/step-registry/additional-enabled-capabilities/additional-enabled-capabilities-commands.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Use yq to create cluster monitoring config, as other steps may adjust it
+YQ_URI=https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64
+YQ_HASH=e70e482e7ddb9cf83b52f5e83b694a19e3aaf36acf6b82512cbe66e41d569201
+echo "${YQ_HASH} -" > /tmp/sum.txt
+if ! curl -Ls "${YQ_URI}" | tee /tmp/yq | sha256sum -c /tmp/sum.txt >/dev/null 2>/dev/null; then
+  echo "Expected file at ${YQ_URI} to have checksum ${YQ_HASH} but instead got $(sha256sum </tmp/yq | cut -d' ' -f1)"
+  strings /tmp/yq
+  exit 1
+fi
+echo "Downloaded yq; sha256 checksum matches expected ${YQ_HASH}."
+chmod +x /tmp/yq
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+if [ -z "${ADDITIONAL_CAPABILITY_SET}" ]; then
+    echo "Not set envrionment variable ADDITIONAL_CAPABILITY_SET, exiting"
+    exit 1
+fi
+
+if [ -z "${BASELINE_CAPABILITY_SET}" ]; then
+    echo "additionalEnabledCapabilities extends the additional capabililities beyoind what you set in baselineCapabilitySet, please also set envrionment variable BASELINE_CAPABILITY_SET, exiting"
+    exit 1
+fi
+
+#select one optional operator from ${additional_caps_set} for testing
+additional_caps_set=(${ADDITIONAL_CAPABILITY_SET})
+additional_caps=${#additional_caps_set[@]}
+selected_optional_operator_index=$((RANDOM % ${additional_caps}))
```

Comment:
An open question here, do we need random select additional caps here:

Advantage:
*  less profile more test coverage.

Disadvantage:
* Hard to reproduce problem since the configuration is not consistent
* Some configuration may not reasonable, e.g. enable baremetal operator on non-BM cluster .

Personally, I prefer the fixed additional caps. @jiajliu @gpei @shellyyang1989 WDYT?

Reply:
Good question! With the growing of additional caps list, there will be new optional cap added, maybe random will not guarantee the new added one be picked. IMO, if the optional operators are independently of each other, how about `None + all additional caps` to cover each cap in this matrix for `None + additionalEanbledCapabilites`? And leave `None + one of/ two of caps` in our own case to reduce the ci profiles here.

Reply:
yes, I think it's also about what exactly do we want prow to do for us? 
`Cover all check points`? vs. `Cover basic functionality is working`?

Here I would prefer the suggestion from Yunfei and JiaJia, a fixed additional caps list should help to guarantee the `additionalEanbledCapabilites ` works as expected. 

As to enable `all additional caps ` or `one or two caps`, either is ok for me.

Reply:
Thanks all for your suggestion, a fixed additional caps are more reasonable. As Jia's suggestion, optional operators are independent of each other at least now, I would change to `None + all additional caps` if no objections. If there is any special case in the feature, we can add new workflow to cover that.